dockstore-webservice/src/main/java/io/dockstore/webservice/core/Doi.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.core;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.media.Schema.RequiredMode;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.persistence.UniqueConstraint;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+@Entity
+@Schema(description = "A Digital Object Identifier (DOI)")
+@Table(name = "doi", uniqueConstraints = @UniqueConstraint(name = "unique_doi_name", columnNames = { "name" }))
+public class Doi {
+    // DOI order of precedence of greatest to least
+    public static final List<DoiCreator> DOI_ORDER_OF_PRECEDENCE = List.of(DoiCreator.USER, DoiCreator.GITHUB, DoiCreator.DOCKSTORE);
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Schema(description = "Implementation specific ID for the DOI in this web service")
+    private long id;
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.STRING)
+    @Schema(description = "The type of DOI", requiredMode = RequiredMode.REQUIRED)
+    private DoiType type;
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.STRING)
+    @Schema(description = "The DOI creator", requiredMode = RequiredMode.REQUIRED)
+    private DoiCreator creator;
+
+    @Column(nullable = false)
+    @Schema(description = "The DOI name", requiredMode = RequiredMode.REQUIRED)
```

Comment:
can add examples to openapi, I think this is the `10.1080/10509585.2015.1092083` bit?

Reply:
Yeah, I'll add an example. I think the existing DOI column calls it `doiUrl` which is a little misleading because it's not a URL. Figured `name` was more appropriate based on the description on [doi.org](https://www.doi.org/)

> A DOI name is a digital identifier of an object, any object — physical, digital, or abstract. DOIs solve a common problem: keeping track of things. Things can be matter, material, content, or activities.